doc/man3/OSSL_HTTP_parse_url.pod

```diff
@@ -0,0 +1,73 @@
+=pod
+
+=head1 NAME
+
+OSSL_HTTP_parse_url,
+OCSP_parse_url
+- http utility functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/http.h>
+
+ int OSSL_HTTP_parse_url(const char *url,
+                         int *pssl, char **puser, char **phost,
+                         char **pport, int *pport_num,
+                         char **ppath, char **pquery, char **pfrag);
```

Comment:
As a general parser, I can see one piece missing; I think it would be interesting for the caller to know the scheme too.  People who want to write OSSL_STORE backends that handle all sorts of schemes (whadoIknow, someone might come up with an LDAP implementation, for example) could find that useful.  In that light, I do question the general use of `*pssl`, as that's probably better handled by the caller, who should know better exactly which scheme they are interested in.

Reply:
All right, done.
This was again much more effort than expected dues to the nastiness of C strings.

Reply:
> dues to the nastiness of C strings.

Eh gods, tell me about it.